actix-files/src/service.rs

```diff
@@ -127,11 +121,19 @@ impl Service<ServiceRequest> for FilesService {
                     Ok(resp) => ok(resp),
                     Err(err) => ok(ServiceResponse::from_err(err, req)),
                 })
-            } else {
-                Box::pin(ok(ServiceResponse::from_err(
+            };
+
+            match self.index {
+                Some(ref index) => match NamedFile::open(path.join(index)) {
+                    Ok(named_file) => serve_named_file(req, named_file),
+                    Err(_) if self.show_index => show_index(req),
+                    Err(err) => self.handle_err(err, req),
+                },
+                None if self.show_index => show_index(req),
+                _ => Box::pin(ok(ServiceResponse::from_err(
                     FilesError::IsDirectory,
                     req.into_parts().0,
```

Comment:
Although unrelated, but I think that this should be `self.handle_err(FilesError::IsDirectory, req)` to give a chance for the `::default_handler(_)`, if set, to run?